benchmarks/bench_algos.py

```diff
@@ -0,0 +1,169 @@
+from collections import OrderedDict
+import pytest
+
+import cudf
+import cugraph
+
+import pytest_benchmark
+# FIXME: Remove this when rapids_pytest_benchmark.gpubenchmark is available
+# everywhere
+try:
+    from rapids_pytest_benchmark import setFixtureParamNames
+except ImportError:
+    print("\n\nWARNING: rapids_pytest_benchmark is not installed, "
+          "falling back to pytest_benchmark fixtures.\n")
+
+    # if rapids_pytest_benchmark is not available, just perfrom time-only
+    # benchmarking and replace utils with nops
+    gpubenchmark = pytest_benchmark.plugin.benchmark
+    def setFixtureParamNames(*args, **kwargs):
+        pass
+
+###############################################################################
+# Utilities
+#
+# FIXME: move utilities to a more reusable location/module
+def getEdgelistFromCsv(csvFileName, delim=' '):
+    """
+    Returns a cuDF DataFrame containing the columns read in from
+    csvFileName. Optional delim string defaults to ' ' (space) for CSV reading.
+    """
+    cols = ["src", "dst", "val"]
+    dtypes = OrderedDict([
+            ("src", "int32"),
+            ("dst", "int32"),
+            ("val", "float32"),
+            ])
+
+    gdf = cudf.read_csv(csvFileName, names=cols, delimiter=delim,
+                        dtype=list(dtypes.values()))
+
+    if gdf['src'].null_count > 0:
+        raise RuntimeError("The reader failed to parse the input")
+    if gdf['dst'].null_count > 0:
+        raise RuntimeError("The reader failed to parse the input")
+    # Assume an edge weight of 1.0 if dataset does not provide it
+    if gdf['val'].null_count > 0:
+        gdf['val'] = 1.0
+    return gdf
+
+
+def getGraphFromEdgelist(edgelistGdf, createDiGraph=False,
+                         renumber=False, symmetrized=False):
+    """
+    Returns a cugraph Graph or DiGraph object from edgelistGdf. renumber and
+    symmetrized can be set to True to perform those operation on construction.
+    """
+    if createDiGraph:
+        G = cugraph.DiGraph()
+    else:
+        G = cugraph.Graph(symmetrized=symmetrized)
+    G.from_cudf_edgelist(edgelistGdf, source="src",
+                         destination="dst", edge_attr="val",
+                         renumber=renumber)
+    return G
```

Comment:
not sure the utility of this function.  Within you if block, just do a cugraph.from_cudf_edgelist( and set the create_using flag